src/invert.ts

```diff
@@ -0,0 +1,49 @@
+import { purry } from './purry';
+
+type Inverted<T extends object> = T[keyof T] extends PropertyKey
+  ? Record<T[keyof T], keyof T>
+  : never;
+
+/**
+ * Returns an object whose keys are values are swapped. If the object contains duplicate values,
+ * subsequent values will overwrite previous values.
+ * @param object the object
+ * @signature
+ *    R.invert(object)
+ * @example
+ *    R.invert({ a: "d", b: "e", c: "f" }) // => { d: "a", e: "b", f: "c" }
+ * @data_first
+ * @category object
+ * @pipeable
+ */
+export function invert<T extends object>(object: T): Inverted<T>;
+
+/**
+ * Returns an object whose keys are values are swapped. If the object contains duplicate values,
+ * subsequent values will overwrite previous values.
+ * @param object the object
+ * @signature
+ *    R.invert()(object)
+ * @example
+ *    R.pipe({ a: "d", b: "e", c: "f" }, R.invert()); // => { d: "a", e: "b", f: "c" }
+ * @data_last
+ * @category object
+ * @pipeable
+ */
+export function invert<T extends object>(): (object: T) => Inverted<T>;
```

Comment:
(i think) you can move the type parameter to the returned function, it would make invert's data-last method a little more resilient to weird typing situations. I might be wrong though... i'm not sure.
```suggestion
export function invert(): <T extends object>(object: T) => Inverted<T>;
```